### PR TITLE
feat: Add support for t-shirt-sized deployments

### PR DESCRIPTION
**Pull request by kimjyhello**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,8 +113,11 @@ No resources.
...
 | <a name="input_other_wandb_env"></a> [other\_wandb\_env](#input\_other\_wandb\_env) | Extra environment variables for W&B | `map(string)` | `{}` | no |
+| <a name="input_redis_reserved_ip_range"></a> [redis\_reserved\_ip\_range](#input\_redis\_reserved\_ip\_range) | Reserved IP range for REDIS peering connection | `string` | `"10.30.0.0/16"` | no |
+| <a name="input_redis_tier"></a> [redis\_tier](#input\_redis\_tier) | Specifies the tier for this Redis instance | `string` | `"STANDARD_HA"` | no |
 | <a name="input_resource_limits"></a> [resource\_limits](#input\_resource\_limits) | Specifies the resource limits for the wandb deployment | `map(string)` | <pre>{<br>  "cpu": null,<br>  "memory": null<br>}</pre> | no |
 | <a name="input_resource_requests"></a> [resource\_requests](#input\_resource\_requests) | Specifies the resource requests for the wandb deployment | `map(string)` | <pre>{<br>  "cpu": "2000m",<br>  "memory": "2G"<br>}</pre> | no |
+| <a name="input_size"></a> [size](#input\_size) | Deployment size for the instance | `string` | `null` | no |
```
Comment (wandb-zacharyblasczyk) [inline README.md:120]: @yogeshg, was this supposed to be just `size` or did I mishear you in the last sync call we had?

File: examples/public-dns-with-cloud-dns/main.tf [hunk 2]
```diff
@@ -23,10 +23,10 @@ module "wandb" {
...
 
   allowed_inbound_cidrs = var.allowed_inbound_cidrs
-  namespace   = var.namespace
-  license     = var.license
-  domain_name = var.domain_name
-  subdomain   = var.subdomain
+  namespace             = var.namespace
+  license               = var.license
+  domain_name           = var.domain_name
+  subdomain             = var.subdomain
```
Comment (wandb-zacharyblasczyk) [inline examples/public-dns-with-cloud-dns/main.tf:29]: Spacing?

File: modules/redis/variables.tf [hunk 3]
```diff
@@ -17,4 +17,14 @@ variable "network" {
 variable "memory_size_gb" {
   description = "The amount of memory which will be allocated to the Redis instance; this value must be expressed in gibibytes."
   type        = number
+}
+
+variable "reserved_ip_range" {
+  description = "This defines the IP range for REDIS peering connection"
+  type        = string
+}
```
Comment (wandb-zacharyblasczyk) [inline modules/redis/variables.tf:25]: Do we want to set the default here?

File: examples/public-dns-with-cloud-dns/variables.tf [hunk 4]
```diff
@@ -79,3 +79,9 @@ variable "disable_code_saving" {
   description = "Boolean indicating if code saving is disabled"
   default     = false
 }
+
+variable "size" {
```
Comment (yogeshg) [inline examples/public-dns-with-cloud-dns/variables.tf:83]: https://github.com/wandb/terraform-google-wandb/pull/91/files#r1486697627
I am expecting the variable to be `size`. @wandb-zacharyblasczyk

File: README.md (hunk 1 above)
Comment (wandb-zacharyblasczyk) [reply README.md:120]: Actually, I think this is the only instance of the var.

Comment (yogeshg) [reply README.md:120]: `input_size` is just the link name, the input variable is indeed called `size`, and that's what im expecting. lmk if that answers / clarifies your question.

File: modules/redis/variables.tf (hunk 3 above)
Comment (wandb-zacharyblasczyk) [reply modules/redis/variables.tf:25]: Ahh, I see, we are defaulting it elsewhere. I think we should keep the defaults incase someone is using the submodule outside of our main.tf.

**Review by gls4 — APPROVED** [review]

**Comment by jsbroks** [comment]
This PR is included in version 1.23.0 :tada:

File: outputs.tf [hunk 5]
```diff
@@ -69,11 +69,21 @@ output "url" {
   description = "The URL to the W&B application"
 }
 
+output "standardized_size" {
+  value = var.size
+}
 
+output "gke_node_count" {
+  value = local.deployment_size[var.size].node_count
```
Comment (yogeshg) [inline outputs.tf:77]: these need try too

**Pull request merged**